Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -1,163 +1,137 @@
-import demistomock as demisto
+"""Implementation file for IllumioCore Integration."""
+
 import urllib3
+from illumio import PolicyComputeEngine, ServicePort, convert_protocol
+from illumio.explorer import TrafficQuery
+from illumio.policyobjects import *
+
+import demistomock as demisto
 from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
 from CommonServerUserPython import *  # noqa
-from illumio import PolicyComputeEngine, convert_protocol, ServicePort
-from illumio.policyobjects import ServicePort, VirtualService
-from illumio.explorer import TrafficQuery
 
 urllib3.disable_warnings()
 
-''' CONSTANTS '''
+""" CONSTANTS """
 
 MIN_PORT = 1
 MAX_PORT = 65535
 HR_DATE_FORMAT = "%d %b %Y, %I:%M %p"
-SOCKET_PREFIX = "IPPROTO_"
-VALID_PROTOCOLS = ['tcp', 'udp']
-VALID_POLICY_DECISIONS = ['potentially_blocked', 'blocked', 'unknown', 'allowed']
-
-'''EXCEPTION CLASS'''
-
-
-class InvalidPortException(Exception):
-    """Exception class for Invalid port."""
-
-    def __init__(self, port):
-        self.message = "{} is an invalid value for port. Value must be in 1 to 65535.".format(port)
-        super().__init__(self.message)
+VALID_PROTOCOLS = ["tcp", "udp"]
+VALID_POLICY_DECISIONS = ["potentially_blocked", "blocked", "unknown", "allowed"]
 
+"""EXCEPTION CLASS"""
 
-class InvalidSingleSelectException(Exception):
-    """Exception class for invalid single-select option."""
 
-    def __init__(self, value, argument, argument_list):
-        self.message = "{} is an invalid value for {}. Possible values are: {}.".format(value, argument, argument_list)
-        super().__init__(self.message)
-
-
-class InvalidMultiSelectException(Exception):
-    """Exception class for invalid multi-select option."""
+class InvalidValueError(Exception):
+    """Custom exception class for invalid values."""
+    def __init__(self, arg_name, arg_value, arg_list=[], message=""):
+        if message:
+            self.message = message.format(arg_value, arg_name)
```

Comment:
This implementation is bad. How are callers expected to know that the message will be formatted with two values? If I called this function with `message="foo"` the exception itself would raise an exception. Just make the arg params optional so the message can be passed as-is.

You should also be calling `super().__init__(message)` here.

Reply:
I'll make the changes

Reply:
Made the changes.